std-bits/table/src/main/java/org/enso/table/aggregations/AggregateColumn.java

```diff
@@ -0,0 +1,96 @@
+package org.enso.table.aggregations;
+
+import org.enso.table.data.table.problems.AggregatedProblems;
+import org.enso.table.data.table.problems.Problem;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/***
+ * Interface used to define aggregate columns.
+ */
+public abstract class AggregateColumn {
+  private final String name;
+  private final int type;
+  private AggregatedProblems problems;
+
+  protected AggregateColumn(String name, int type) {
+    this.name = name;
+    this.type = type;
+    this.problems = null;
+  }
+
+  /***
+   * @return Name of the new column.
+   */
+  public final String getName() {
+    return name;
+  }
+
+  /***
+   * @return The type of the new column.
+   */
+  public int getType() {
+    return type;
+  }
+
+  public AggregatedProblems getProblems() {
+    return problems;
+  }
+
+  /***
+   * Compute the value for a set of rows
+   * @param rows - indexes to the rows in the source table to aggregate on
+   * @return aggregated value
+   */
+  public Object aggregate(int[] rows) {
```

Comment:
```suggestion
  public Object aggregate(long[] rows) {
```

I'm pretty sure our numbers are `long` by default, I don't remember seeing `int` in such contexts.

Reply:
Also if we have a specialization for integer columns to avoid boxing, may be worth to also have one for `double` columns?

Reply:
these are the indexes - a bad name of the variable.